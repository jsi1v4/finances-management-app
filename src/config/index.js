export const EN_US = 'en';
export const PT_BR = 'pt';
export const INIT_LOCALE = EN_US;
export const ICON =
  'https://raw.githubusercontent.com/jsi1v4/jsi1v4/master/assets/favicon.ico';
export const LOGO =
  'https://raw.githubusercontent.com/jsi1v4/jsi1v4/master/assets/logo192.png';

export default {
  INIT_LOCALE,
  EN_US,
  PT_BR,
  ICON,
  LOGO,
};
