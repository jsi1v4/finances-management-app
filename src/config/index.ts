import { Locale } from 'src/i18n/types';

export const INIT_LOCALE = Locale.EN_US;
export const ICON =
  'https://raw.githubusercontent.com/jsi1v4/jsi1v4/master/assets/favicon.ico';
export const LOGO =
  'https://raw.githubusercontent.com/jsi1v4/jsi1v4/master/assets/logo192.png';
