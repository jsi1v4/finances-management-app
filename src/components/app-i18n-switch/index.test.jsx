import React from 'react';
import { render } from '@testing-library/react';

import { EN_US } from '../../config';
import { I18nSwitch } from './index';

describe('I18nSwitch component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<I18nSwitch />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render component', async () => {
    const handleClick = jest.fn();
    const { getByText } = render(
      <I18nSwitch value={EN_US} onChange={handleClick} />
    );
    expect(getByText('English').innerHTML).toBeDefined();
    expect(getByText('PortuguÃªs').innerHTML).toBeDefined();
  });
});
