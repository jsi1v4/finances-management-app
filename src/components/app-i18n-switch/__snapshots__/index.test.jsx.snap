// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`I18nSwitch component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-radio-group ant-radio-group-outline"
  >
    <label
      class="ant-radio-button-wrapper"
    >
      <span
        class="ant-radio-button"
      >
        <input
          class="ant-radio-button-input"
          type="radio"
          value="en"
        />
        <span
          class="ant-radio-button-inner"
        />
      </span>
      <span>
        English
      </span>
    </label>
    <label
      class="ant-radio-button-wrapper"
    >
      <span
        class="ant-radio-button"
      >
        <input
          class="ant-radio-button-input"
          type="radio"
          value="pt"
        />
        <span
          class="ant-radio-button-inner"
        />
      </span>
      <span>
        PortuguÃªs
      </span>
    </label>
  </div>
</DocumentFragment>
`;
